### PR TITLE
[goto_check] Avoid shift check on divisions and modulo operator

### PR DESCRIPTION
**Pull request by brcfarias**
This PR addresses the issues found during the execution of svcomp hardware benchmark as detailed at https://github.com/esbmc/esbmc/issues/978#issuecomment-1526557969

### CODE REVIEW

**Review by lucasccordeiro — APPROVED** [review]

File: src/goto-programs/goto_check.cpp [hunk 1]
```diff
@@ -394,6 +394,9 @@ void goto_checkt::shift_check(
   if(!enable_ub_shift_check)
     return;
 
+  if(!is_lshr2t(expr) && !is_ashr2t(expr) && !is_shl2t(expr))
```
Comment (rafaelsamenezes) [inline src/goto-programs/goto_check.cpp]: Why are we arriving at this function with anything besides a shift? Shouldn't we:

- Add an assertion that this function is only reachable within shift operations
- Fix the flow that is calling a shift check on something that it shouldn't

**Review by rafaelsamenezes — CHANGES_REQUESTED** [review]

**Comment by lucasccordeiro** [comment]
@brcfarias: could you please address @rafaelsamenezes's comments here?

File: src/goto-programs/goto_check.cpp (hunk 1 above)
Comment (brcfarias) [reply src/goto-programs/goto_check.cpp]: >Why are we arriving at this function with anything besides a shift

We didn't have a break in the switch-case of goto_checkt::check_rec(...) after case expr2t::modulus_id, so shift_check was being called on this operation. Initially, I preferred a minimal change to reduce the chances of side effects, but I agree with you that it is better to adjust the flow, and it seems safe to modify it. I will update the commit. Thank you for the suggestion.

**Review by rafaelsamenezes — APPROVED** [review]
LGTM

**Comment by lucasccordeiro** [comment]
Thanks, both!

As soon as CI is green, I'll merge this PR.

**Pull request merged**